internal/databases/mongo/binary/storage_local.go

```diff
@@ -101,21 +100,21 @@ func (localStorage *LocalStorage) EnsureEmptyDBPath() error {
 	}
 
 	for _, name := range names {
-		if !deletionProtectionWhitelist.MatchString(name) {
+		if localStorage.whitelist == nil || !localStorage.whitelist.MatchString(name) {
 			return fmt.Errorf("directory '%v' is not empty", localStorage.MongodDBPath)
 		}
 	}
 
 	return nil
 }
 
-func CreateWhiteList() *regexp.Regexp {
-	val, ok := conf.GetSetting(conf.MongoDBDeletionProtectionWhitelist)
-	exp, err := regexp.Compile(val)
+func CreateWhiteList(dirPath string) *regexp.Regexp {
+	filesRegexp, ok := conf.GetSetting(conf.MongoDBDeletionProtectionWhitelist)
+	_, err := regexp.Compile(filesRegexp)
 
 	if !ok || err != nil {
-		return regexp.MustCompile(`^lost\+found$`)
+		filesRegexp = `lost\+found`
 	}
 
-	return exp
+	return regexp.MustCompile(filepath.Join(regexp.QuoteMeta(dirPath), filesRegexp))
```

Comment:
for me it bad decision , add parent folder like prefix 
1) this is not obvious and the user can add at the beginning ^ , then everything breaks down
2) can't realise reason why need it prefix

Reply:
So, should we declare full path regexp here? Or should we declare relative path and use regexp.MatchString? 
First one would be too large to be simple imho and second one also could break down easily with ^

Reply:
I guess we need to check, as before, just the name(suffix after directory path) . You changed this logic because of IsDirectoryEmpty ,  but we can cut prefix before check with whiteListRegexp.
`relativePath := strings.TrimPrefix(strings.Cut(path, directoryPath), '/') ` after this check `relativePath`